### ISSUE TITLE
Update u64 module in stdlib to use u32assert2 operation

### ISSUE THREAD

**Issue by bobbinth** [maticnetwork/miden#217]
Currently, [u64 module](https://github.com/maticnetwork/miden/blob/next/stdlib/asm/math/u64.masm) in the standard library, when executing "checked" versions of procedures, verifies that inputs are u64 values using `u32assert4` procedure. This procedure uses `u32assert` assembly instruction on limbs of input values. This is rather inefficient and can be improved.

First trivial improvement is to modify `u32assert4` procedure to use `u32assert2` operations internally. In fact, this would make it equivalent to `u32assertw` assembly instruction.

Another improvement could be to actually modify "checked" versions of procedures to minimize the number of stack manipulation operations. For example, `checked_add` could be written as:
```
swap
movup.3
u32assert.2
u32add.unsafe
movup.3
movup.3
u32assert.2
u32addc.unsafe
eq.0
assert
```
This should be a few cycles faster than using `u32assertw` upfront to check if all 4 limbs are u32 values.

**Comment by DeimosTheMisthios** [comment]
I can take this one, too.

**Comment by bobbinth** [comment]
@DeimosTheMisthios  Awesome! Thank you!

**Comment by bobbinth** [comment]
@DeimosTheMisthios - are you still planning to work on this issue?

**Comment by bobbinth** [comment]
Closed by #345
